src/funcx_common/task_storage/base.py

```diff
@@ -0,0 +1,67 @@
+import abc
+import typing as t
+
+from ..tasks import TaskProtocol
+
+
+class StorageException(Exception):
+    def __init__(self, reason: str):
+        self.reason = reason
+
+    def __str__(self) -> str:
+        return f"Storage request failed due to reason:{self.reason}"
+
+
+class TaskStorage(abc.ABC):
+    """
+    Abstract class which defines the interface for task storage.
+    """
+
+    @abc.abstractmethod
+    def store_result(self, task: TaskProtocol, result: str) -> bool:
+        """
+        Store the result of a task.
+
+        If the storage call succeeded, set reference to data in Task.result_reference
+        and return True
+        If the call failed, an exception would be raised.
+        """
+
+    @property
+    @classmethod
+    @abc.abstractmethod
+    def storage_id(cls) -> str:
+        raise NotImplementedError()
+
+    @property
+    def backward_compatible(cls) -> bool:
+        """
+        Indicates backward compatibility with Task from v0.3.3 and older where
+        Task.result is used without Task.result_reference
+        :return:
+        Bool
+        """
+        return False
```

Comment:
Rather than a property, why not just a class attribute?
```
class TaskStorage:
    backward_compatible: bool = False
```

Reply:
I don't imagine this to be in use for more than a couple of weeks after deploying. If this doesn't break anything, we should leave this as is to be scuttled in a 2+ weeks time when we remove the bits for backward compatibility.

Reply:
That doesn't change the fact that this is eight lines of code declaring an instance property, which misleadingly declares its argument as `cls` (it is not a class property), which could be a single line assignment?

I don't see the expected longevity of the code to be a rationale for writing something which is more complicated and not-quite-incorrect but also not-quite-correct.